services/ui-src/src/components/layout/Header.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import { RouterWrappedComponent } from "utils/testing/setupJest";
+import { Header, MenuOption } from "components";
+import { testA11y } from "utils/testing/commonTests";
+
+const headerComponent = (
+  <RouterWrappedComponent>
+    <Header handleLogout={() => {}} />
+  </RouterWrappedComponent>
+);
+
+describe("<Header />", () => {
+  describe("Test Header", () => {
+    beforeEach(() => {
+      render(headerComponent);
+    });
+
+    test("Header is visible", () => {
+      const header = screen.getByRole("navigation");
+      expect(header).toBeVisible();
+    });
+
+    test("Logo is visible", () => {
+      expect(screen.getByAltText("HCBS logo")).toBeVisible();
+    });
+
+    test("Help button is visible", () => {
+      expect(screen.getByAltText("Help")).toBeVisible();
+    });
+
+    test("Menu button is visible", () => {
+      expect(screen.getByAltText("Arrow down")).toBeVisible();
+    });
+
+    test("Renders My Account and is clickable", () => {
+      render(
+        <MenuOption
+          text={"My Account"}
+          icon={"icon_arrrow_down.png"}
+          altText={"Arrow down"}
+        />
+      );
+
+      const menuButton = screen.getByRole("button", { name: /my account/i });
+      expect(menuButton).toBeInTheDocument();
+
+      fireEvent.click(menuButton);
```

Comment:
It's a good idea to use `userEvent` instead of `fireEvent` pretty much everywhere. `fireEvent` is a more low-level interface, whereas `userEvent` does a better job of simulating actual user flow.

It is in a separate package*, so if HCBS doesn't already have `@testing-library/user-event`, this would be a good time to install it.

*For now. The author (of both libraries) has said he intends to eventually include `userEvent` in the main package. For this and more, see his blog: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-testing-libraryuser-event

```suggestion
      userEvent.click(menuButton);
```

Reply:
This test is also missing any kind of assertion about what happens when the menu button is clicked. Will that cause a navigation? We can detect those in unit tests. Will it cause another element to become visible? We can detect that too.

Reply:
Thank you for that! While researching how to make test more effective, I ran into the `fireEvent` but `userEvent` is definitely the better option here.